tests/test_simple.py

```diff
@@ -1,5 +1,48 @@
+"""Simple expression that should pass with mypy."""
+import operator
+
 import numpy as np
+from typing import Iterable
+
+# Basic checks
+array = np.array([1, 2])
+def ndarray_func(x: np.ndarray) -> np.ndarray:
+    return x
+ndarray_func(np.array([1, 2]))
+array == 1
+array.dtype == float
+
+# Iteration and indexing
+def iterable_func(x: Iterable) -> Iterable:
+    return x
+iterable_func(array)
+[element for element in array]
+iter(array)
+zip(array, array)
+array[1]
+array[:]
+array[...]
+array[:] = 0
+
+array_2d = np.ones((3, 3))
+array_2d[:2, :2]
+array_2d[..., 0]
+array_2d[:2, :2] = 0
+
+# Other special methods
+len(array)
+str(array)
+array + 1
+-array
 
-def foo(a: np.ndarray): pass
+array_scalar = np.array(1)
+int(array_scalar)
+float(array_scalar)
+complex(array_scalar)
```

Comment:
mypy current errors on this line: `tests/test_simple.py:41: error: No overload variant of "complex" matches argument types [numpy.ndarray]`

Is this a mypy bug or am I doing something wrong?

Reply:
I believe this is https://github.com/python/mypy/issues/2136, in other words, you need to add `SupportsComplex` as a base since mypy doesn't yet infer protocols based on dunder methods.

Reply:
Reported at https://github.com/python/mypy/issues/4592